File: java/src/n02_test_data_generator/DataGeneratorCSV.java
package n02_test_data_generator;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.StringJoiner;

public class DataGeneratorCSV extends DataGenerator {
    @Override
    void saveData(File file) {
        StringJoiner content = new StringJoiner(",");
        for (Double dataPoint : this.dataSet) {
            content.add(dataPoint.toString());
        }
        try (FileWriter writer = new FileWriter(file)) {
            writer.write(content.toString());
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: dart/src/n02_test_data_generator/data_generator_csv.dart
import 'dart:io';

import 'data_generator.dart';

class DataGeneratorCSV extends DataGenerator {
  @override
  Future<void> saveData(final File file) async {
    await file.writeAsString('${this.dataSet.join(',')}${Platform.lineTerminator}');
  }
}
